lwt/cohttp_lwt_unix.ml

```diff
@@ -75,7 +75,9 @@ module Server = struct
       Lwt_stream.on_terminate stream (fun () ->
         ignore_result (Lwt_io.close ic));
       let body = Cohttp_lwt_body.of_stream stream in
-      let res = Cohttp.Response.make ~status:`OK ~encoding ?headers () in
+      let mime_type = Magic_mime.lookup fname in
+      let headers = Cohttp.Header.add_opt headers "content-type" mime_type in
```

Comment:
Agreed. I think it's a useful feature, as long as it's only for respond_file and that it doesn't override any headers.